app/gosqs/gosqs.go

```diff
@@ -373,9 +373,17 @@ func ReceiveMessage(w http.ResponseWriter, req *http.Request) {
 	loops := waitTimeSeconds * 10
 	for loops > 0 {
 		app.SyncQueues.RLock()
-		found := len(app.SyncQueues.Queues[queueName].Messages)-numberOfHiddenMessagesInQueue(*app.SyncQueues.Queues[queueName]) != 0
+		_, queueFound := app.SyncQueues.Queues[queueName]
+		var messageFound bool
+		if queueFound {
+			messageFound = len(app.SyncQueues.Queues[queueName].Messages)-numberOfHiddenMessagesInQueue(*app.SyncQueues.Queues[queueName]) != 0
+		}
 		app.SyncQueues.RUnlock()
-		if !found {
+		if !queueFound {
```

Comment:
Move this above the `var messageFound bool` so to exit earlier on error.

Reply:
go idea moved it up to exit earlier, thanks!